pkg/gofr/container/datasources.go

```diff
@@ -199,3 +199,15 @@ type HealthChecker interface {
 	// It is done to avoid adding packages which are not being used.
 	HealthCheck(context.Context) (any, error)
 }
+
+type KVStore interface {
+	Get(ctx context.Context, key string) (string, error)
+	Set(ctx context.Context, key string, value string) error
+	Delete(ctx context.Context, key string) error
+}
+
+type KVStoreProvider interface {
+	KVStore
+
+	provider
```

Comment:
Maybe you could add an HealthChecker, by using any random badger function

maybe

https://pkg.go.dev/github.com/dgraph-io/badger/v4#DB.IsClosed

I might be commenting at the wrong place as this is the KVStore interface